tests/aws/services/lambda_/test_lambda_api.py

```diff
@@ -760,6 +760,191 @@ def test_function_arns(
+            {
+                "args": {
+                    "FunctionName": f"arn:aws:lambda:invalid-region:{account_id}:function:my-function"
+                    + "-"
+                    + "a" * max_arn_length,
+                    "Qualifier": "a" * max_qualifier_length,
+                },
+                "SnapshotName": "full_arn_and_qualifier_too_long_and_invalid_region",
+            },
+            {
+                "args": {
+                    "FunctionName": "my-function",
+                    "Qualifier": "a" * max_qualifier_length,
+                },
+                "SnapshotName": "qualifier_too_long",
+            },
+            {
+                "args": {
+                    "FunctionName": f"arn:aws:lambda:{region_name}:{account_id}:function:my-function:1:2"
+                },
+                "SnapshotName": "full_arn_with_multiple_qualifiers",
+            },
+            {
+                "args": {"FunctionName": "my-function", "Qualifier": "-1"},
+                "SnapshotName": "negative_version_number",
+                "SkipCases": [
+                    "create_function"
+                ],  # CreateFunction: Valid start token for a qualifier
+            },
+            {
+                "args": {"FunctionName": f"arn:aws:lambda:{region_name}:{account_id}:function"},
+                "SnapshotName": "incomplete_arn",
+                "SkipCases": [
+                    "create_function"
+                ],  # CreateFunction: Valid ARN where FunctionName=function
+            },
+            {
+                "args": {"FunctionName": "function:my-function:$LATEST:extra"},
+                "SnapshotName": "partial_arn_with_extra_qualifier",
+            },
+            {
+                "args": {
+                    "FunctionName": f"arn:aws:lambda:{region_name}:{account_id}:function:my-function:$LATEST",
+                    "Qualifier": "1",
+                },
+                "SnapshotName": "latest_version_with_additional_qualifier",
+            },
+            {
+                "args": {
+                    "FunctionName": f"arn:aws:lambda:{region_name}:{account_id}:function:my-function",
+                    "Qualifier": "$latest",
+                },
+                "SnapshotName": "lowercase_latest_qualifier",
+            },
+            {
+                "args": {"FunctionName": f"arn:aws:lambda::{account_id}:function:my-function"},
+                "SnapshotName": "missing_region_in_arn",
+            },
+            {
+                "args": {"FunctionName": f"arn:aws:lambda:{region_name}::function:my-function"},
+                "SnapshotName": "missing_account_id_in_arn",
+            },
+            {
+                "args": {
+                    "FunctionName": f"arn:aws:lambda:{region_name}:{account_id}:function:my-function:$LATES"
+                },
+                "SnapshotName": "misspelled_latest_in_arn",
+            },
+        ]
+
+        for test in tests:
```

Comment:
Could we also extract this into parametrization?

Reply:
Yeah sure. I was concerned about the test snapshot naming being quite ugly though since the params are added to the snapshot naming, which is why I kept the test-table internal as opposed to parametrizing it.

Think it's a good suggestion though

Reply:
Take a look at the `ids` parameter of the parametrization, you could use the snapshot name in there, which should alleviate this concern.

Reply:
Defined these in pytest fixtures 😄